components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -234,10 +234,11 @@ func (r *NotebookReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 					oldConditions[0].Message != newCondition.Message {
 					log.Info("Appending to conditions: ", "namespace", instance.Namespace, "name", instance.Name, "type", newCondition.Type, "reason", newCondition.Reason, "message", newCondition.Message)
 					instance.Status.Conditions = append([]v1beta1.NotebookCondition{newCondition}, oldConditions...)
-				}
```

Comment:
@mofanke I am pretty sure the check on `#222` which is `if pod.Status.ContainerStatuses[i].State == instance.Status.ContainerState {` will mean that we always need to update the status if we get past line `#222`.

You have made it so that the status will only be updated if the conditions array is updated, I think this will mean the container state is never updated.

Reply:
yes, thanks ,my mistake.

I founded error msg :{"level":"error","ts":1625132033.5322628,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"notebook","request":"flow/mofanke162","error":"Notebook.kubeflow.org \"mofanke162\" is invalid: status.conditions: Invalid value: \"null\": status.conditions in body must be of type array: \"null\"","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:218\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88"}


so i changed update  status when condition update, the error is gone. but as you said ,  the container state is never updated..